Issue #207 : Styling of kbd tags

Siddharth1010 opened the pull request:
Signed-off-by: Siddharth <siddharthmani2000@gmail.com>

Fixes: #207 Added CSS to the kbd tags

spoenemann reviewed (commented):
Thanks for the contribution!

Have you tested this with dark theme? The website should switch to dark when you change your OS theme settings. I tried to test it because I guess the bright `kbd` items don't look very nice in dark theme, but I couldn't find an extension that has `kbd` tags in its README. Do you know one?

If possible, it would be better to use the colors given by the Material UI theme instead of fixed values.

spoenemann commented:
You can access the theme colors with the `theme` parameter given by the `createStyles` function.

Siddharth1010 commented:
@spoenemann No I found a few keyboard commands defined under the 'code' tag. (Overview section in 'Material Theme')

With regards to the colours, shall I change it to the default "theme.palette.neutral.light/dark" as defined in the file?

Siddharth1010 commented:
I used the Inspect Element tab on Chrome, changed the tag from 'code' to 'kbd' and added the current style. This is what it looks like
![image](https://user-images.githubusercontent.com/59433969/104467702-a4253500-55dc-11eb-95d2-b7bcea116546.png)

Please let me know if I should change the color and maybe reduce the size?

spoenemann commented:
Thanks for the screenshot! It does break the general dark color concept. Maybe we should use the same background color as for code?

I don't know how "theme.palette.neutral.light/dark" looks like in that theme.

Siddharth1010 commented:
As you suggested I have made the colors similar to the code tag for dark mode. It looks like this:
![image](https://user-images.githubusercontent.com/59433969/104563347-89e86700-566f-11eb-8c03-fb6ddc42f7fc.png)

spoenemann commented:
Thanks for the hints. Now it looks like this for me:
<img width="198" alt="Screen Shot 2021-01-14 at 11 32 09" src="https://user-images.githubusercontent.com/4067210/104579398-37517f80-565c-11eb-87df-2101d9ca8f34.png">

Why is the foreground color different from your screenshot?

If you don't find a theme color that has enough contrast and differs from `code` tags, you could use something like this:
```
color: theme.palette.type === 'dark' ? theme.palette.grey[300] : theme.palette.grey[700],
```

Another alternative could be to remove the `color` style and change the font:
```
fontFamily: theme.typography.fontFamily,
fontSize: theme.typography.fontSize - 1
```

Other issues:
 * I would remove `cursor: 'pointer'` as that suggests that clicking on those elements does something.
 * `boxShadow` also has a hard-coded color, which does not work in dark theme. Please use a theme color instead. I guess the MUI theme offers access to shadow colors, but I don't know where.

Sorry for the nagging, CSS is often an endless story 😅

Siddharth1010 commented:
No that's alright I am happy to contribute :)
I have removed the color style and added fonts as you have specified. I have removed the cursor style. 
I was reading the MUI documentation for Shadow and it seems to be useless unless defined inside a 'Box' element. So I have removed that as well. Please let me know if this is okay :)

spoenemann reviewed (approved):
Looks good now, thanks!

spoenemann commented:
This is not deployed to open-vsx.org automatically, but we'll do it at the next opportunity.

Siddharth1010 commented:
@spoenemann  Thank you so much. I look forward to making more contributions :)

I was hoping to apply to the eclipse foundation for GSoC 2021. I would be really grateful if you could give me some advice regarding the same.

spoenemann commented:
I am not working for the Eclipse Foundation, so I cannot help with that. But have you seen this?
https://wiki.eclipse.org/Google_Summer_of_Code

Siddharth1010 commented:
No I hadn't. Thanks a lot :)
